File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        N=len(nums)
        maxN=nums[0]
        if N<2: return True
        for i in range(len(nums)):
            maxN= max(maxN,i+ nums[i])
            
            if i>=maxN:
                if i ==N-1: return True
                return False
        return True

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int N= nums.length;
        int maxN = nums[0];
        
        if(N<2) return true;
        for (int i = 0; i < N; i++){
            maxN = Math.max(maxN, (i+nums[i]));
            if(i>=maxN){
                if(i==N-1) return true;
                return false;
            }
        }
        return true;
    }
}